sdks/java/core/src/test/java/org/apache/beam/sdk/io/CompressedSourceTest.java

```diff
@@ -738,7 +1069,133 @@ public void testGzipProgress() throws IOException {
       assertThat(readerOrig, instanceOf(CompressedReader.class));
       CompressedReader<Byte> reader = (CompressedReader<Byte>) readerOrig;
       // before starting
-      assertEquals(0.0, reader.getFractionConsumed(), 1e-6);
+      assertEquals(0.0, reader.getFractionConsumed(), DELTA);
+      assertEquals(0, reader.getSplitPointsConsumed());
+      assertEquals(1, reader.getSplitPointsRemaining());
+
+      // confirm has three records
+      for (int i = 0; i < numRecords; ++i) {
+        if (i == 0) {
+          assertTrue(reader.start());
+        } else {
+          assertTrue(reader.advance());
+        }
+        assertEquals(0, reader.getSplitPointsConsumed());
+        assertEquals(1, reader.getSplitPointsRemaining());
+      }
+      assertFalse(reader.advance());
+
+      // after reading empty source
```

Comment:
```suggestion
      // after reading source
```

Reply:
line 325: public void testReadConcatenatedLzo() throws IOException:
Is this for LZOP codec? Since unlike LZOP, LZO Codec supports file concatenation. Also we have testReadMultiStreamLzo and testFalseReadConcatenatedLzop

Reply:
It would be better if the concatenated streams for LZOP worked or if concatenated streams were detected then an exception was thrown to the user.

Reply:
For now we have added a comment warning users that a concatenated lzo file doesn't gets decompressed correctly. Its added above testFalseReadConcatenatedLzop and testFalseReadMultiStreamLzop methods.

Reply:
Best to add that comment over the LZOP enum since nobody reading the documentation is going to find the comment in the tests.